packages/mattermost-redux/src/utils/helpers.ts

```diff
@@ -82,22 +83,7 @@ export const isMinimumServerVersion = (currentVersion: string, minMajorVersion =
 
 // Generates a RFC-4122 version 4 compliant globally unique identifier.
 export function generateId(): string {
-    // implementation taken from http://stackoverflow.com/a/2117523
-    let id = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
-    id = id.replace(/[xy]/g, (c) => {
-        const r = Math.floor(Math.random() * 16);
-        let v;
-
-        if (c === 'x') {
-            v = r;
-        } else {
-            // eslint-disable-next-line no-mixed-operators
-            v = r & 0x3 | 0x8;
-        }
-
-        return v.toString(16);
-    });
-    return id;
+    return crypto.randomUUID();
```

Comment:
@hmhealey Can you give me some advice please?
`insecure randomness` CodeQL alert is reported and the reason is the code below.
(This code change is in https://github.com/mattermost/mattermost-webapp/pull/7723)
```ts
const r = Math.floor(Math.random() * 16);
```
I checked the annotation of the [implementation](http://stackoverflow.com/a/2117523) and used the alternative(`crypto.randomUUID`).
Is there a reason why the `uuid` library or `crypto` module wasn't used?
If so, I would find other alternatives.

Reply:
Please revert this for now. If CodeQL complains more, then we can silence its errors either manually or by submitting a separate PR specifically to update this. I need to reach out to our security team to figure out if they had some plan for addressing these since they're disrupting unrelated PRs like this one.

As for this specific case, I think it's just very old code that keeps getting copied around, so we never bothered making it more secure. All of the important IDs are generated by the server, so we only use this code for tests and in a couple places where the client needs a temporary ID while the server is generating a real one

Reply:
Thanks for the advice. I reverted the related commits.